python/hail/dataset.py

```diff
@@ -2326,6 +2326,33 @@ def grm(self):
         jkm = self._jvdf.grm()
         return KinshipMatrix(jkm)
 
+    @handle_py4j
+    @record_method
+    @typecheck(key_expr=strlike,
+               agg_expr=strlike,
+               single_key=bool)
+    def group_variants_by(self, key_expr, agg_expr, single_key=True):
+        """Group variants by key, aggregating along sample.
+        
+        **Examples**
+
+        Compute the max depth of each sample on each contig:
+
+        >>> vds = vds.group_variants_by("v.contig", "gs.map(g => g.dp).max()")
+
+        :param str key_expr: Expression for new column key.
+
+        :param str agg_expr: Expression for aggregating along samples.
```

Comment:
when we improve docs, it will be important to explain the scope of key_expr and agg_expr, and the allowed form of agg_expr